xbmc/Application.cpp

```diff
@@ -3909,6 +3925,11 @@ bool CApplication::OnMessage(CGUIMessage& message)
       g_infoManager.SetCurrentItem(*m_itemCurrentFile);
```

Comment:
I think additional changes are needed here now, because setCurrentItem will notify observers about a changed file, which after your changes will not be set, since you no longer call "setCurrentMovie" which updated m_currentFile of GUIInfomanager. So all this method now does is to reset m_currentFile of the Infomanger and notify observers, while no new data is available yet for them to consume. So I suppose that this call will become somewhat obsolete and the observers would need to be notified on CGUIInfomanger::updateInfo() instead?
I'm not too familiar with how the Infomanager works, so I could be wrong on the impact of triggering the observers while no data is yet available in Infomanager.